server/server.go

```diff
@@ -150,11 +150,44 @@ func (s *Server) StartEtcd(apiHandler http.Handler) error {
 	s.client = client
 	s.id = uint64(etcd.Server.ID())
 
+	if err = s.initClusterID(); err != nil {
+		return errors.Trace(err)
+	}
+	log.Infof("init cluster id %v", s.clusterID)
+
+	s.rootPath = path.Join(pdRootPath, strconv.FormatUint(s.clusterID, 10))
+	s.idAlloc = &idAllocator{s: s}
+	s.cluster = newRaftCluster(s, s.clusterID)
+
 	// Server has started.
 	atomic.StoreInt64(&s.closed, 0)
 	return nil
 }
 
+func (s *Server) initClusterID() error {
+	// Get any cluster key to parse the cluster ID.
+	resp, err := kvGet(s.client, pdRootPath, clientv3.WithLastCreate()...)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	// If we can get any key except for "pdClusterIDPath", we need to
+	// parse the existed cluster ID from the key for compatibility.
+	if len(resp.Kvs) > 0 {
+		if key := string(resp.Kvs[0].Key); key != pdClusterIDPath {
```

Comment:
what I mean is you can use the value directly, no need call next initAndGet function.